pkg/dependencies/factory.go

```diff
@@ -54,6 +57,8 @@ type Dependencies struct {
 	DignosticCollectorFactory diagnostics.DiagnosticBundleFactory
 	CAPIManager               *clusterapi.Manager
 	ResourceSetManager        *clusterapi.ResourceSetManager
+	FileReader                *files.Reader
```

Comment:
why do we need FileReader as a dependency?

Reply:
`FileReader` is a dependency for `ManifestReader`
It also gives the flexibility to only build by itself if needed

Reply:
We are using it to build `ManifestReader`